src/ui/navigation/container/bottomBar.tsx

```diff
@@ -0,0 +1,231 @@
+
+  const [AnimatedHiding] = useState(new Animated.Value(0))
+
+  useEffect(() => {
+    AnimatedHiding.setValue(0)
+
+    if (navigation) {
+      const blurListener = navigation.addListener('willBlur', animateHiding)
+      const focusListener = navigation.addListener('didFocus', animateAppear)
+
+      const isIOS = Platform.OS === 'ios'
+
+      let keyboardShowListener: EmitterSubscription,
+        keyboardHideListener: EmitterSubscription
+
+      if (isIOS) {
+        keyboardShowListener = Keyboard.addListener(
+          'keyboardWillShow',
+          animateAppear,
+        )
+        keyboardHideListener = Keyboard.addListener(
+          'keyboardWillHide',
+          animateHiding,
+        )
+      } else {
+        keyboardShowListener = Keyboard.addListener(
+          'keyboardDidShow',
+          animateAppear,
+        )
+        keyboardHideListener = Keyboard.addListener(
+          'keyboardDidHide',
+          animateHiding,
+        )
+      }
+
+      return () => {
+        blurListener.remove()
+        focusListener.remove()
+        keyboardShowListener.remove()
+        keyboardHideListener.remove()
+      }
+    }
+
+    return
+  }, [])
+
+  const animateHiding = () => {
+    Animated.timing(AnimatedHiding, {
+      // NOTE: to account for screens that use SafeAreaView
+      toValue: BAR_HEIGHT - 2 * VERTICAL_OFFSET,
+      duration: 400,
+      useNativeDriver: true,
+    }).start()
+  }
+
+  const animateAppear = () => {
+    Animated.timing(AnimatedHiding, {
+      toValue: 0,
+      duration: 100,
+      useNativeDriver: true,
+    }).start()
+  }
+
+  return (
+    <AnimatedSafeAreaView
+      style={[styles.wrapper, { transform: [{ translateY: AnimatedHiding }] }]}
+      forceInset={safeAreaInset}
+    >
+      <View style={[styles.buttonWrapper]}>
+        {routes.map((route, i) => {
+          const focused = i === index
+          const scene = { route, index, focused }
+          const label = getLabelText(scene) as string
+          const testID = getTestID(scene)
+          const accessibility = {
+            label: getAccessibilityLabel(scene),
+            role: getAccessibilityRole(scene) as AccessibilityRole,
+            states: getAccessibilityStates(scene) as AccessibilityState[],
+          }
+
+          return (
+            <React.Fragment key={route.key}>
+              {i === 2 && (
+                <View style={{ width: BUTTON_SIZE + VERTICAL_OFFSET }} />
+              )}
+              <TabButton
+                testID={testID}
+                scene={scene}
+                renderIcon={renderIcon}
+                onTabPress={() => onTabPress(scene)}
+                label={label}
+                accessibility={accessibility}
+                colors={colors}
+              />
+            </React.Fragment>
+          )
+        })}
+      </View>
+      <InteractionButton
+        topMargin={VERTICAL_OFFSET}
+        buttonSize={BUTTON_SIZE}
+        scale={SCREEN_PROPORTION}
```

Comment:
Not sure if necessary to send the values as props. Did it this way since all the calculations were done in the container.